docs/faqs.md

```diff
@@ -0,0 +1,153 @@
+
+- Often, porting known state-of-the-art compilation techniques to apply to structured programs can
+  be highly non-trivial.
+
+- There is tension between maximal optimization and compact program representation. There are still
+  a lot of questions we need to consider and resolve when it comes to compiling dynamic, structured
+  programs. For example, the more compact the program representation (which is great for
+  compilation scalability!) the less information we potentially have up-front to maximally optimize
+  programs at compile time. How do we deal with such scenarios?
+
+## How did you decide what goes into the schema?
+
+## Why Cap'n Proto?
+
+We use [Cap'n Proto](https://capnproto.org/) as our data serialization format for Jeff.
+
+**Reasons for choosing Cap'n Proto:**
+- Fast, efficient binary encoding
+- Zero-copy deserialization for high performance
+- Simple and expressive schema language
+- Robust tooling and support for key quantum programming languages (Python, C++, Rust)
+
+**Alternative serialization formats:**  
+Other models considered include [Protocol Buffers](https://protobuf.dev/), [Flatbuffers](https://flatbuffers.dev/), [MessagePack](https://msgpack.org/), and JSON. While these formats offer similar features, Cap'n Proto provides the best combination of speed, schema support, and language integration for our needs.
+
+
+## Isn't this just *another* format standard?
+
+The danger of introducing new exchange formats or standards is that they themselves may slow down
+development — perhaps because of a governance structure with too much process and/or stakeholders,
+or simply because the format is too rigid and doesn't permit extensibility for new ideas.
+
+This is something we wish to avoid — Jeff is *not* a standard we need all stakeholders to adopt, but
+an exchange format localized to compilers, designed to
+be easily extensible. Additional information for program representation and compilation can be added
+directly, and will be taken into account by software frameworks that understand the direction.
+Further, development will happen out in the open, on GitHub.
+
+## Why not format X?
+
+### Why not OpenQASM2?
+
+[OpenQASM2](https://arxiv.org/abs/1707.03429) has become the de-facto interchange format in quantum software, but the biggest drawback
+has been the lack of program structure — something that we *need* to be a first-class citizen in a
+compiler exchange format.
+
+An exchange format that makes it easy to incorporate structure will also allow the field to consider
+compilation of structured programs by default.
+
+### Why not OpenQASM3?
+
+[OpenQASM3](https://arxiv.org/abs/2104.14722) brings program structure to OpenQASM2, however has not yet seen adoption or tooling
+maturity in the ecosystem to the extent as OpenQASM2. In addition, OpenQASM3 maintains a focus on
+human readability (requiring string generation and parsing), and is not designed first and foremost for quantum optimization.
```

Comment:
I think the two main disadvantages should be placed in the bulleted list for better visibility. To me these are the reliance on parsing, and the linear arrangement of gates rather than an arrangement in a graph.

Reply:
https://github.com/unitaryfoundation/jeff/pull/10/files#r2312562198